### PR TITLE
Serve fonts from origin instead of cdn

### LINKED ISSUE #154: Serve the docs fonts ourselves
For the MVP of the documentation, fonts were served directly from Google's CDNs. The ideal scenario would be to serve the fonts ourselves.

### PR DESCRIPTION
**Pull request by Harsh-br0**
## 📚 Description

fixes #154

## 🔖 Changes

- added a `fonts.css` to link cached font files
- all cached fonts lies inside `docs/public/fonts` directory

## ✅ To-do list

- [x] Make sure that all the pipeline passes
- [x] Make sure you have updated the documentation to reflect the changes or new features
- [x] Make sure to update the `CHANGELOG.md` to reflect the new feature or fix

### CODE REVIEW

**Review by Tito-Kati — CHANGES_REQUESTED** [review]
Hello, thank you very much for helping us with the project, would you mind making some changes to simplify it?

Since the documentation is only in English, we do not need the versions with the unicode-ranges, the standard version of each font will be sufficient, you can download them from Google Fonts.

Make sure you download the correct weights for each font, we need 400 for Cousine, 600 for Rubik, and for Heebo we need both 400 and 600.

Additionally, it would be great if the filenames of the font files have a name that can be understood.

Thank you very much again for helping us.

**Comment by Tito-Kati** [comment]
I also noticed that you've updated the yarn.lock, please @Harsh-br0 restore it to the previous version as we have it there precisely to ensure we all have the same versions of the packages and to ensure that the documentation works stably.

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,10 @@
 .idea/
 .env
 
+# package manager
+package-lock.json
+yarn.lock
```
Comment (Chemaclass) [inline .gitignore:7]: We don't want to ignore these lock files

Comment (Harsh-br0) [reply .gitignore:7]: oh, let me fix it again

**Comment by Harsh-br0** [comment]
Alright, my apologies for responding lately
so i have covered all the requested changes now

**Comment by Harsh-br0** [comment]
Seems like my formatter messed up line endings

File: docs/.vitepress/theme/style.css [hunk 2]
```diff
@@ -7,16 +7,25 @@
...
  */
 
-@import url('https://fonts.googleapis.com/css2?family=Heebo:wght@400;600&display=swap');
-@import url('https://fonts.googleapis.com/css2?family=Cousine');
-@import url('https://fonts.googleapis.com/css2?family=Rubik:wght@600');
+/**
+ * @import url('https://fonts.googleapis.com/css2?family=Heebo:wght@400;600&display=swap');
+ * @import url('https://fonts.googleapis.com/css2?family=Cousine');
+ * @import url('https://fonts.googleapis.com/css2?family=Rubik:wght@600');
+ */
```
Comment (Chemaclass) [inline docs/.vitepress/theme/style.css:14]: Are these lines needed? Why not removing them?

Comment (Harsh-br0) [reply docs/.vitepress/theme/style.css:14]: I kept them as a reference, but there seems no issue in removing them

Comment (Chemaclass) [reply docs/.vitepress/theme/style.css]: There is no need to keep commented code, especially if you're using **git** as it's a source version control and we can easily see the before/after and get to the removed code whenever needed (if needed) 😄

**Comment by Chemaclass** [comment]
Can you please check the To-do list from the PR itself? 
At least the third box: "updating the CHANGELOG" is something you can also add to this PR :)

Screenshot 2023-10-08 at 18 09 34

**Comment by Harsh-br0** [comment]
so, i have checked first 2 checkboxes which makes sense to this PR
But , should i commit the commit messages to `CHANGELOG.md` then the 3rd checkbox would be valid to this PR ?

**Comment by Chemaclass** [comment]
@Harsh-br0, what I would suggest is to add  only one line like "Serve fonts from origin instead of cdn" to the Unreleased list from the CHANGELOG. That should be enough

**Comment by Tito-Kati** [comment]
I believe that since it's not a significant improvement for our users, there's no need to add anything to the changelog, but it would be a shame if your name didn't appear in the next release.

Perhaps you could add something like "optimized docs fonts".

**Comment by Harsh-br0** [comment]
Alright, Now it seems fine

**Review by Tito-Kati — APPROVED** [review]
Superb! Thanks you!

**Comment by Harsh-br0** [comment]
Nice, Thanks for such amazing library.
It was good to work with you

**Pull request merged**